Add financial distress

igelstorm opened the pull request:
## What

- Add financial distress variable to the initial population data (dichotomous variable defined as 4 or 5 for UKHLS variable [finnow](https://www.understandingsociety.ac.uk/documentation/mainstage/variables/finnow/))
- Add process to update financial distress (see below for model specification)
- Add code to include ethnicity in simulation
  - This is needed as a predictor for financial distress, and was already in initial population data but not yet used

## Why

Resolves #181.

## To do before merging

- [x] Update training data to include financial distress
- [x] Update expected output for integration test

## Model specification for financial distress process

Logit model estimated in UKHLS. Predictors:

- Economic variables:
  - Employment transition
  - Working hours
  - Change in log income
  - Income decrease (binary)
  - Poverty transition
  - Gross personal non-employment capital income (L1)
  - Gross personal private pension income (L1)
- Lagged/fixed confounders
  - Receives benefits (L1)
  - Owns home (L1)
  - Partnership status (L1)
  - Number of children (L1)
  - SF-12 PCS (L1)
  - SF-12 MCS (L1)
  - Region (L1)
  - HH income quintile (L1)
  - Long-term sick/disabled (L1)
  - Financial distress (L1)
  - Gender
  - Age (L1)
  - Age^2 (L1)
  - Education
  - Ethnicity
  - Year

igelstorm commented:
> This code seems good! Failing validation tests - is that likely caused by different initial population?

I thought so at first, but I tried to keep the initial population unchanged except for the financial distress variable, and there are still small differences.

I think because I've added a process that involves additional random draws, all the subsequent random draws will be different, leading to slightly different random noise in the output. I have to say that diminishes the value of that test a bit. But maybe we can be content with verifying that the output statistics remain _approximately_ the same?

matteorichiardi commented:
...something to discuss with @Van De Ven, Justin W ***@***.***>
 perhaps?

On Fri, 13 Jun 2025 at 09:47, Andrew Baxter ***@***.***>
wrote:

> *andrewbaxter439* left a comment (centreformicrosimulation/SimPaths#198)
> <https://github.com/centreformicrosimulation/SimPaths/pull/198#issuecomment-2969575252>
>
> I had wondered if the additional draws were partly to blame, though these
> don't seem to affect the previous draws in the same Innovations object:
>
> class InnovationsTest {
>
>     @Test
>     public void checkSame() {
>
>         Innovations innovation1 = new Innovations(2, 100);
>         Innovations innovation2 = new Innovations(3, 100);
>
>
>         assertTrue(innovation1.getDoubleDraw(0) == innovation2.getDoubleDraw(0) );
>         assertFalse(innovation1.getDoubleDraw(1) == innovation2.getDoubleDraw(2) );
>
>     }
>
> }
>
> ... passes fine?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/centreformicrosimulation/SimPaths/pull/198#issuecomment-2969575252>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOK4OAFPD7UVKW6JSKQDHL3DKFZ3AVCNFSM6AAAAAB65GG3VWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNRZGU3TKMRVGI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

andrewbaxter439 commented on src/main/java/simpaths/model/SimPathsModel.java:
```diff
@@ -584,6 +584,9 @@ public void buildSchedule() {
 
 
+        // Update financial distress
+        addCollectionEventToAllYears(persons, Person.Processes.FinancialDistress);
```
Brief note - I'd suggest putting these before the MCS/PCS/Mental health stage 2 estimates. Financial distress as a product of employment and income changes would be a causal mediator of effects on mental health?

igelstorm commented:
@andrewbaxter439 I've updated this based on discussions:

- Still using 5 ethnicity categories
- Including non-employment capital income and private pension income as covariates
- Using cross-sectional weights (`dimxwt`)
- Updated description of covariates to reflect lags correctly

andrewbaxter439 commented:
Awesome that this is working - think ready to merge I'd say!
